parser-typechecker/src/Unison/Builtin/Decls.hs

```diff
@@ -760,7 +763,7 @@ forceTerm :: (Var v) => a -> a -> Term v a -> Term v a
 forceTerm a au e = Term.app a e (unitTerm au)
 
 delayTerm :: (Var v) => a -> Term v a -> Term v a
-delayTerm a = Term.lam a $ Var.named "()"
+delayTerm a = Term.lam a $ Var.typed Var.Delay
```

Comment:
We now use a `Var.typed Var.Delay` to for variable names that correspond to the argument to a thunk.